src/FiberTraces.cc

```diff
@@ -2683,62 +2683,49 @@ namespace pfsDRPStella = pfs::drp::stella;
+    void assignITrace( FiberTraceSet< ImageT, MaskT, VarianceT > & fiberTraceSet,
                        ndarray::Array< T, 1, I > const& fiberIds,
-                       ndarray::Array< U, 1, I > const& xCenters,
-                       ndarray::Array< U, 1, I > const& yCenters ){
-      int iTrace;
-      int startPos = 0;
+                       ndarray::Array< U, 1, I > const& xCenters ){
+      size_t iTrace;
+      size_t startPos = 0;
       const ndarray::Array< T, 1, 1 > fiberTraceIds = pfs::drp::stella::math::unique( fiberIds );
-      ndarray::Array< int, 1, 1 > thisFiberTrace = pfs::drp::stella::math::where( fiberIds, "==", fiberTraceIds[ 0 ], int(1), int(0) );
-      int nRows = std::accumulate( thisFiberTrace.begin(), thisFiberTrace.end(), 0 );
+      ndarray::Array< size_t, 1, 1 > thisFiberTrace = pfs::drp::stella::math::where( fiberIds, "==", fiberTraceIds[ 0 ], size_t(1), size_t(0) );
+      size_t nRows = std::accumulate( thisFiberTrace.begin(), thisFiberTrace.end(), 0 );
       for (size_t i = 0; i < fiberTraceSet.size(); ++i) {
         PTR( FiberTrace<ImageT, MaskT, VarianceT> ) fiberTrace = fiberTraceSet.getFiberTrace( i );
         iTrace = findITrace( *fiberTrace,
                              xCenters,
-                             yCenters,
                              fiberTraceIds.getShape()[ 0 ],
                              nRows,
                              startPos );
         startPos = iTrace + 1;
-        cout << "FiberTraces::assignITrace: i = " << i << ": iTrace = " << iTrace << endl;
-        if ( iTrace < 0 ){
-          cout << "FiberTraces::assignITrace: i = " << i << ": trace not found => returning FALSE" << endl;
-          return false;
-        }
+        #ifdef __DEBUG_ASSIGNITRACE__
+          cout << "FiberTraces::assignITrace: i = " << i << ": iTrace = " << iTrace << endl;
+        #endif
         fiberTrace->setITrace( iTrace );
       }
-      return true;
+      return;
     }
 
     template< typename ImageT, typename MaskT, typename VarianceT, typename U, int I >
-    int findITrace( FiberTrace< ImageT, MaskT, VarianceT > const& fiberTrace,
-                    ndarray::Array< U, 1, I > const& xCentersArr,
-                    ndarray::Array< U, 1, I > const& yCentersArr,
-                    int nTraces,
-                    int nRows,
-                    int startPos ){
-      #ifdef __DEBUG_FINDITRACE__
-        cout << "FiberTraces::findITrace: fiberTraceIds = " << fiberTraceIds << endl;
-      #endif
-      float xCenter;
-      unsigned int yCenter;
-      xCenter = fiberTrace.getFiberTraceFunction()->xCenter;
-      yCenter = fiberTrace.getFiberTraceFunction()->yCenter;
+    size_t findITrace( FiberTrace< ImageT, MaskT, VarianceT > const& fiberTrace,
+                       ndarray::Array< U, 1, I > const& xCentersArr,
+                       size_t nTraces,
+                       size_t nRows,
+                       size_t startPos ){
```

Comment:
Why are we changing all of these to `size_t`? I understand that in your new scheme you can't return a negative, but presumably you're not going to overflow `INT_MAX` anyway. If you want to be formally correct, shouldn't it be an NDArray `size_type` rather than `size_t`?